when excute cmd "cmake --build . --config Release",raise error

**the device as follow:**
NVIDIA GeForce RTX 3080,ubuntu22.04,cuda_11.5

**the detail of the problem as follow:**
(gavin) lucy@pop-os:~/test0710/rwkv.cpp$ cmake --build . --config Release
Consolidate compiler generated dependencies of target ggml
[ 75%] Built target ggml
Consolidate compiler generated dependencies of target rwkv
[ 78%] Built target rwkv
Consolidate compiler generated dependencies of target test_ggml_basics
[ 80%] Built target test_ggml_basics
Consolidate compiler generated dependencies of target test_quantized_matmul_on_gpu
[ 81%] Linking C executable ../bin/test_quantized_matmul_on_gpu
/usr/bin/ld: ../ggml/src/libggml.a(ggml-cuda.cu.o): in function `ggml_backend_cuda_split_buffer_type_alloc_buffer(ggml_backend_buffer_type*, unsigned long)':
tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0x217): undefined reference to `operator new(unsigned long)'
/usr/bin/ld: ../ggml/src/libggml.a(ggml-cuda.cu.o): in function `ggml_cuda_log(ggml_log_level, char const*, ...)':
tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0x989): undefined reference to `operator new(unsigned long)'
/usr/bin/ld: tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0xa01): undefined reference to `operator delete(void*)'
/usr/bin/ld: ../ggml/src/libggml.a(ggml-cuda.cu.o): in function `ggml_backend_cuda_buffer_free_buffer(ggml_backend_buffer*)':
tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0x1399): undefined reference to `operator delete(void*)'
/usr/bin/ld: ../ggml/src/libggml.a(ggml-cuda.cu.o): in function `ggml_backend_cuda_split_buffer_free_buffer(ggml_backend_buffer*)':
tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0x1478): undefined reference to `operator delete(void*)'
/usr/bin/ld: tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0x1497): undefined reference to `operator delete(void*)'
/usr/bin/ld: ../ggml/src/libggml.a(ggml-cuda.cu.o): in function `ggml_backend_cuda_free(ggml_backend*)':
tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0x16e4): undefined reference to `operator delete(void*)'
/usr/bin/ld: ../ggml/src/libggml.a(ggml-cuda.cu.o):tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0x16f6): more undefined references to `operator delete(void*)' follow
/usr/bin/ld: ../ggml/src/libggml.a(ggml-cuda.cu.o): in function `ggml_backend_cuda_event_new(ggml_backend*)':
tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0x212f): undefined reference to `operator new(unsigned long)'
/usr/bin/ld: ../ggml/src/libggml.a(ggml-cuda.cu.o): in function `ggml_backend_cuda_buffer_type_alloc_buffer(ggml_backend_buffer_type*, unsigned long)':
tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0x21f1): undefined reference to `operator new(unsigned long)'
/usr/bin/ld: tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0x22b8): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_construct(unsigned long, char)'
/usr/bin/ld: tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0x2339): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_replace(unsigned long, unsigned long, char const*, unsigned long)'
/usr/bin/ld: tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0x238a): undefined reference to `operator delete(void*)'
/usr/bin/ld: ../ggml/src/libggml.a(ggml-cuda.cu.o): in function `ggml_cuda_info()':
tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0x276b): undefined reference to `__cxa_guard_acquire'
/usr/bin/ld: tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0x2786): undefined reference to `__cxa_guard_release'
/usr/bin/ld: ../ggml/src/libggml.a(ggml-cuda.cu.o): in function `ggml_backend_cuda_context::new_pool_for_device(int)':
tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0x2a23): undefined reference to `operator new(unsigned long)'
/usr/bin/ld: tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0x2a8e): undefined reference to `operator new(unsigned long)'
/usr/bin/ld: ../ggml/src/libggml.a(ggml-cuda.cu.o): in function `ggml_backend_cuda_host_buffer_type':
tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0x4be2): undefined reference to `__cxa_guard_acquire'
/usr/bin/ld: tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0x4c34): undefined reference to `__cxa_guard_release'
/usr/bin/ld: ../ggml/src/libggml.a(ggml-cuda.cu.o): in function `ggml_backend_cuda_buffer_type':
tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0x53c9): undefined reference to `operator delete(void*)'
/usr/bin/ld: tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0x53fc): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_construct(unsigned long, char)'
/usr/bin/ld: tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0x5425): undefined reference to `operator new(unsigned long)'
/usr/bin/ld: tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0x5444): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_replace(unsigned long, unsigned long, char const*, unsigned long)'
/usr/bin/ld: tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0x54bd): undefined reference to `std::__throw_system_error(int)'
/usr/bin/ld: ../ggml/src/libggml.a(ggml-cuda.cu.o): in function `ggml_cuda_mul_mat_id(ggml_backend_cuda_context&, ggml_tensor*)':
tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0x899c): undefined reference to `operator new(unsigned long)'
/usr/bin/ld: tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0x8ee4): undefined reference to `operator delete(void*)'
/usr/bin/ld: tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0x98ee): undefined reference to `std::__throw_length_error(char const*)'
/usr/bin/ld: ../ggml/src/libggml.a(ggml-cuda.cu.o): in function `ggml_backend_cuda_init':
tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0xa271): undefined reference to `operator new(unsigned long)'
/usr/bin/ld: tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0xa313): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_construct(unsigned long, char)'
/usr/bin/ld: tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0xa3a4): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_replace(unsigned long, unsigned long, char const*, unsigned long)'
/usr/bin/ld: tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0xa3f2): undefined reference to `operator delete(void*)'
/usr/bin/ld: tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0xa4c6): undefined reference to `operator new(unsigned long)'
/usr/bin/ld: ../ggml/src/libggml.a(ggml-cuda.cu.o): in function `ggml_backend_cuda_split_buffer_init_tensor(ggml_backend_buffer*, ggml_tensor*)':
tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0xb155): undefined reference to `operator new(unsigned long)'
/usr/bin/ld: ../ggml/src/libggml.a(ggml-cuda.cu.o): in function `ggml_backend_cuda_split_buffer_type':
tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0xb7fb): undefined reference to `__cxa_guard_acquire'
/usr/bin/ld: tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0xb857): undefined reference to `__cxa_guard_release'
/usr/bin/ld: tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0xb88f): undefined reference to `operator new(unsigned long)'
/usr/bin/ld: tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0xb8fe): undefined reference to `std::__throw_system_error(int)'
/usr/bin/ld: ../ggml/src/libggml.a(ggml-cuda.cu.o): in function `ggml_backend_cuda_buffer_free_buffer(ggml_backend_buffer*)':
tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0x13a2): undefined reference to `operator delete(void*)'
/usr/bin/ld: ../ggml/src/libggml.a(ggml-cuda.cu.o): in function `ggml_backend_cuda_split_buffer_free_buffer(ggml_backend_buffer*)':
tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0x14af): undefined reference to `operator delete(void*)'
/usr/bin/ld: ../ggml/src/libggml.a(ggml-cuda.cu.o): in function `ggml_backend_cuda_event_free(ggml_backend_event*)':
tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0x15da): undefined reference to `operator delete(void*)'
/usr/bin/ld: ../ggml/src/libggml.a(ggml-cuda.cu.o): in function `ggml_backend_cuda_free(ggml_backend*)':
tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text+0x1714): undefined reference to `operator delete(void*)'
/usr/bin/ld: ../ggml/src/libggml.a(ggml-cuda.cu.o): in function `ggml_cuda_log(ggml_log_level, char const*, ...) [clone .cold]':
tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text.unlikely+0x7): undefined reference to `operator delete(void*)'
/usr/bin/ld: ../ggml/src/libggml.a(ggml-cuda.cu.o):tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text.unlikely+0x23): more undefined references to `operator delete(void*)' follow
/usr/bin/ld: ../ggml/src/libggml.a(ggml-cuda.cu.o): in function `ggml_cuda_info() [clone .cold]':
tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text.unlikely+0x43): undefined reference to `__cxa_guard_abort'
/usr/bin/ld: ../ggml/src/libggml.a(ggml-cuda.cu.o): in function `ggml_backend_cuda_context::new_pool_for_device(int) [clone .cold]':
tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text.unlikely+0x57): undefined reference to `operator delete(void*)'
/usr/bin/ld: ../ggml/src/libggml.a(ggml-cuda.cu.o): in function `ggml_backend_cuda_host_buffer_type.cold':
tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text.unlikely+0x1a2): undefined reference to `__cxa_guard_abort'
/usr/bin/ld: ../ggml/src/libggml.a(ggml-cuda.cu.o): in function `ggml_backend_cuda_buffer_type.cold':
tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text.unlikely+0x1b5): undefined reference to `operator delete(void*)'
/usr/bin/ld: tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text.unlikely+0x1c7): undefined reference to `operator delete(void*)'
/usr/bin/ld: ../ggml/src/libggml.a(ggml-cuda.cu.o): in function `ggml_cuda_mul_mat_id(ggml_backend_cuda_context&, ggml_tensor*) [clone .cold]':
tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text.unlikely+0x2f3): undefined reference to `operator delete(void*)'
/usr/bin/ld: ../ggml/src/libggml.a(ggml-cuda.cu.o): in function `ggml_backend_cuda_init.cold':
tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text.unlikely+0x314): undefined reference to `operator delete(void*)'
/usr/bin/ld: tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text.unlikely+0x31c): undefined reference to `operator delete(void*)'
/usr/bin/ld: ../ggml/src/libggml.a(ggml-cuda.cu.o):tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text._ZNSt8_Rb_treeISt5arrayIfLm16EESt4pairIKS1_24ggml_backend_buffer_typeESt10_Select1stIS5_ESt4lessIS1_ESaIS5_EE8_M_eraseEPSt13_Rb_tree_nodeIS5_E.isra.0[_ZNSt3mapISt5arrayIfLm16EE24ggml_backend_buffer_typeSt4lessIS1_ESaISt4pairIKS1_S2_EEED5Ev]+0x9e): more undefined references to `operator delete(void*)' follow
/usr/bin/ld: ../ggml/src/libggml.a(ggml-cuda.cu.o): in function `void std::vector<ggml_tensor_extra_gpu*, std::allocator<ggml_tensor_extra_gpu*> >::_M_realloc_insert<ggml_tensor_extra_gpu* const&>(__gnu_cxx::__normal_iterator<ggml_tensor_extra_gpu**, std::vector<ggml_tensor_extra_gpu*, std::allocator<ggml_tensor_extra_gpu*> > >, ggml_tensor_extra_gpu* const&)':
tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text._ZNSt6vectorIP21ggml_tensor_extra_gpuSaIS1_EE17_M_realloc_insertIJRKS1_EEEvN9__gnu_cxx17__normal_iteratorIPS1_S3_EEDpOT_[_ZNSt6vectorIP21ggml_tensor_extra_gpuSaIS1_EE17_M_realloc_insertIJRKS1_EEEvN9__gnu_cxx17__normal_iteratorIPS1_S3_EEDpOT_]+0x10a): undefined reference to `operator new(unsigned long)'
/usr/bin/ld: tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text._ZNSt6vectorIP21ggml_tensor_extra_gpuSaIS1_EE17_M_realloc_insertIJRKS1_EEEvN9__gnu_cxx17__normal_iteratorIPS1_S3_EEDpOT_[_ZNSt6vectorIP21ggml_tensor_extra_gpuSaIS1_EE17_M_realloc_insertIJRKS1_EEEvN9__gnu_cxx17__normal_iteratorIPS1_S3_EEDpOT_]+0x143): undefined reference to `std::__throw_length_error(char const*)'
/usr/bin/ld: ../ggml/src/libggml.a(ggml-cuda.cu.o): in function `std::_Rb_tree<std::array<float, 16ul>, std::pair<std::array<float, 16ul> const, ggml_backend_buffer_type>, std::_Select1st<std::pair<std::array<float, 16ul> const, ggml_backend_buffer_type> >, std::less<std::array<float, 16ul> >, std::allocator<std::pair<std::array<float, 16ul> const, ggml_backend_buffer_type> > >::_M_get_insert_unique_pos(std::array<float, 16ul> const&)':
tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text._ZNSt8_Rb_treeISt5arrayIfLm16EESt4pairIKS1_24ggml_backend_buffer_typeESt10_Select1stIS5_ESt4lessIS1_ESaIS5_EE24_M_get_insert_unique_posERS3_[_ZNSt8_Rb_treeISt5arrayIfLm16EESt4pairIKS1_24ggml_backend_buffer_typeESt10_Select1stIS5_ESt4lessIS1_ESaIS5_EE24_M_get_insert_unique_posERS3_]+0x10c): undefined reference to `std::_Rb_tree_decrement(std::_Rb_tree_node_base*)'
/usr/bin/ld: ../ggml/src/libggml.a(ggml-cuda.cu.o): in function `std::pair<std::_Rb_tree_iterator<std::pair<std::array<float, 16ul> const, ggml_backend_buffer_type> >, bool> std::_Rb_tree<std::array<float, 16ul>, std::pair<std::array<float, 16ul> const, ggml_backend_buffer_type>, std::_Select1st<std::pair<std::array<float, 16ul> const, ggml_backend_buffer_type> >, std::less<std::array<float, 16ul> >, std::allocator<std::pair<std::array<float, 16ul> const, ggml_backend_buffer_type> > >::_M_emplace_unique<std::array<float, 16ul>&, ggml_backend_buffer_type&>(std::array<float, 16ul>&, ggml_backend_buffer_type&)':
tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text._ZNSt8_Rb_treeISt5arrayIfLm16EESt4pairIKS1_24ggml_backend_buffer_typeESt10_Select1stIS5_ESt4lessIS1_ESaIS5_EE17_M_emplace_uniqueIJRS1_RS4_EEES2_ISt17_Rb_tree_iteratorIS5_EbEDpOT_[_ZNSt8_Rb_treeISt5arrayIfLm16EESt4pairIKS1_24ggml_backend_buffer_typeESt10_Select1stIS5_ESt4lessIS1_ESaIS5_EE17_M_emplace_uniqueIJRS1_RS4_EEES2_ISt17_Rb_tree_iteratorIS5_EbEDpOT_]+0x1d): undefined reference to `operator new(unsigned long)'
/usr/bin/ld: tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text._ZNSt8_Rb_treeISt5arrayIfLm16EESt4pairIKS1_24ggml_backend_buffer_typeESt10_Select1stIS5_ESt4lessIS1_ESaIS5_EE17_M_emplace_uniqueIJRS1_RS4_EEES2_ISt17_Rb_tree_iteratorIS5_EbEDpOT_[_ZNSt8_Rb_treeISt5arrayIfLm16EESt4pairIKS1_24ggml_backend_buffer_typeESt10_Select1stIS5_ESt4lessIS1_ESaIS5_EE17_M_emplace_uniqueIJRS1_RS4_EEES2_ISt17_Rb_tree_iteratorIS5_EbEDpOT_]+0xaa): undefined reference to `std::_Rb_tree_insert_and_rebalance(bool, std::_Rb_tree_node_base*, std::_Rb_tree_node_base*, std::_Rb_tree_node_base&)'
/usr/bin/ld: tmpxft_0000a863_00000000-6_ggml-cuda.compute_75.cudafe1.cpp:(.text._ZNSt8_Rb_treeISt5arrayIfLm16EESt4pairIKS1_24ggml_backend_buffer_typeESt10_Select1stIS5_ESt4lessIS1_ESaIS5_EE17_M_emplace_uniqueIJRS1_RS4_EEES2_ISt17_Rb_tree_iteratorIS5_EbEDpOT_[_ZNSt8_Rb_treeISt5arrayIfLm16EESt4pairIKS1_24ggml_backend_buffer_typeESt10_Select1stIS5_ESt4lessIS1_ESaIS5_EE17_M_emplace_uniqueIJRS1_RS4_EEES2_ISt17_Rb_tree_iteratorIS5_EbEDpOT_]+0xcc): undefined reference to `operator delete(void*)'
/usr/bin/ld: ../ggml/src/libggml.a(ggml-cuda.cu.o):(.data.rel.ro._ZTI14ggml_cuda_pool[_ZTI14ggml_cuda_pool]+0x0): undefined reference to `vtable for __cxxabiv1::__class_type_info'
/usr/bin/ld: ../ggml/src/libggml.a(ggml-cuda.cu.o):(.data.rel.ro._ZTI18ggml_cuda_pool_leg[_ZTI18ggml_cuda_pool_leg]+0x0): undefined reference to `vtable for __cxxabiv1::__si_class_type_info'
/usr/bin/ld: ../ggml/src/libggml.a(ggml-cuda.cu.o):(.data.rel.ro._ZTI18ggml_cuda_pool_vmm[_ZTI18ggml_cuda_pool_vmm]+0x0): undefined reference to `vtable for __cxxabiv1::__si_class_type_info'
/usr/bin/ld: ../ggml/src/libggml.a(ggml-cuda.cu.o):(.data.rel.local.DW.ref.__gxx_personality_v0[DW.ref.__gxx_personality_v0]+0x0): undefined reference to `__gxx_personality_v0'
**collect2: error: ld returned 1 exit status
gmake[2]: *** [tests/CMakeFiles/test_quantized_matmul_on_gpu.dir/build.make:106：bin/test_quantized_matmul_on_gpu] 错误 1
gmake[1]: *** [CMakeFiles/Makefile2:219：tests/CMakeFiles/test_quantized_matmul_on_gpu.dir/all] 错误 2
gmake: *** [Makefile:146：all] 错误 2**

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -172,7 +172,6 @@ elseif()
     set(GGML_STANDALONE ON)
 endif()
 
-set(BUILD_SHARED_LIBS OFF)
 if (NOT TARGET ggml)
     add_subdirectory(ggml)
     # ... otherwise assume ggml is added by a parent CMakeLists.txt

--- a/rwkv_graph.inc
+++ b/rwkv_graph.inc
@@ -548,7 +548,7 @@ static struct ggml_tensor * rwkv_att_v6(
         ggml_reshape_1d(ctx, layer.att_time_decay, n_embed)
     );
 
-    w = rwkv_exp(ctx, ggml_neg_inplace(ctx, rwkv_exp(ctx, w)));
+    w = rwkv_exp(ctx, ggml_neg(ctx, rwkv_exp(ctx, w)));
     w = ggml_reshape_4d(ctx, w, 1, head_size, head_count, sequence_length);
 
     // dup is not strictly required; doing it just in case.
@@ -576,9 +576,9 @@ static struct ggml_tensor * rwkv_att_v6(
     x = rwkv_group_norm_eps_64e_minus5(ctx, x, head_count);
     // Convert back to a regular vector.
     x = ggml_reshape_2d(ctx, x, n_embed, sequence_length);
-    x = ggml_add_inplace(
+    x = ggml_add(
         ctx,
-        ggml_mul_inplace(
+        ggml_mul(
             ctx,
             x,
             layer.att_ln_x_weight

--- a/rwkv_operators.inc
+++ b/rwkv_operators.inc
@@ -158,7 +158,7 @@ static void rwkv_groupnorm_impl(
 
 // Element-wise exp(x)
 struct ggml_tensor * rwkv_exp(struct ggml_context * ctx, struct ggml_tensor * x) {
-    return ggml_map_custom1_inplace(ctx, x, rwkv_exp_impl, 1, NULL);
+    return ggml_map_custom1(ctx, x, rwkv_exp_impl, 1, NULL);
 }
 
 // Element-wise 1 - x

